x/stakeibc/keeper/host_zone.go

```diff
@@ -50,22 +52,33 @@ func (k Keeper) GetHostZone(ctx sdk.Context, chain_id string) (val types.HostZon
 	return val, true
 }
 
-// GetHostZoneFromDenom returns a hostZone from the relevant coin denom
-
+// GetHostZoneFromDenom returns a hostZone from the relevant coin denom (denom is ibc/...)
 func (k Keeper) GetHostZoneFromDenom(ctx sdk.Context, denom string) (val types.HostZone, found bool) {
 	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.HostZoneKey))
 	iterator := sdk.KVStorePrefixIterator(store, []byte{})
+	var bval types.HostZone
 
 	defer iterator.Close()
 
+	// hash is the part of denom after ibc/
+	hash := strings.Join(strings.SplitN(denom, "ibc/", -1), "")
```

Comment:
could we add a simple error check that complains if the denom is not prefixed by "ibc"?

Reply:
realize this is somewhat pedantic given the lookup below _should_ fail if it's not valid, but I would err on the side of explicitly checking assumptions in the core liquid staking logic

Reply:
This is a great catch, I think this might expose an attack vector! Per huddle, it sounds like this is taken care of with TEST-50